### PR TITLE
[IMP] contract: Add extend existing invoice possibility

### PR DESCRIPTION
**Pull request by carlosdauden**
@Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I will do it the contrary, taking precedence on the values of the invoice passed as argument, and it lacks to check if the invoice is already validated for example. As it's a WIP that is not going to be finished, I close.

**Comment by pedrobaeza** [comment]
I reopen this as this has been reinitiated. Please take into account my comments and resolve conflicts.

**Comment by carlosdauden** [comment]
Changes done

**Comment by pedrobaeza** [comment]
Travis is red

**Comment by carlosdauden** [comment]
Travis green

File: contract/models/account_analytic_account.py [hunk 1]
```diff
@@ -236,10 +236,30 @@ def _prepare_invoice(self):
         return invoice._convert_to_write(invoice._cache)
 
     @api.multi
-    def _create_invoice(self):
+    def _prepare_invoice_update(self, invoice):
+        vals = self._prepare_invoice()
```
Comment (pedrobaeza) [inline contract/models/account_analytic_account.py:240]: This is still overwriting all the previous values, where they should be more important than new ones.

Comment (carlosdauden) [reply contract/models/account_analytic_account.py:240]: No, this function return update_vals.
Review the function, vals only is taked to respect any changes in _prepare_invoice asignation fields

Comment (pedrobaeza) [reply contract/models/account_analytic_account.py:240]: OK, got it

**Review by pedrobaeza — APPROVED** [review]

**Review by rafaelbn — APPROVED** [review]
Tested in runbot :+1:

**Pull request merged**